src/main/java/org/usrv/http/ClientRequest.java

```diff
@@ -87,6 +85,12 @@ public void validate() {
             throw new InvalidRequestException("Unsupported method: " + this.method);
         } else if (!Objects.equals(this.protocol, "HTTP/1.1")) {
             throw new InvalidRequestException("Unsupported protocol: " + this.protocol);
+        } else if(this.isHostRequiredForProtocol() && !this.headers.containsKey("Host")) {
+            throw new InvalidRequestException(InvalidRequestException.MISSING_REQUIRED_HOST_MESSAGE);
         }
     }
+
+    private boolean isHostRequiredForProtocol(){
+        return Objects.equals(this.protocol, "HTTP/1.1");
+    }
```

Comment:
I like this! Good to have prepared for an eventual fallback to 1.0. However, clients can send HTTP/2 and 3 requests as well (where we would respond with a regular 1.1 response). In those cases, we want to return a 400 response as well.